.github/workflows/firefox-ios-ui-tests.yml

```diff
@@ -207,11 +207,12 @@ jobs:
     run-fullfunctionaltests:
       name: Run full functional tests
       runs-on: macos-14
-      needs: compile
+      needs: run-smoketests
```

Comment:
In the long run, I would like to run the smoke tests first so that no job is blocked due to too many runners in use.